### PR TITLE
[accounting]: rename accountingservice to accounting

### PR DESCRIPTION
**Pull request by puckpuck**
# Changes

Renames `accountingservice` to `accounting`.

This is the first service to get renamed based on the overall [rename](https://github.com/open-telemetry/opentelemetry-demo/issues/1788) effort for all components in the demo.

I did a "Find All" for `accountingservice` and `accounting_service` without case sensitivity and manually updated each entry. I did a full docker prune, then ran `docker compose build accounting` followed by `make start`. Tested in Jaeger and Grafana that all telemetry data is working as intended.

## Merge Requirements

For new features contributions, please make sure you have completed the following
essential items:

* [x] `CHANGELOG.md` updated to document new feature additions
* [ ] Appropriate documentation updates in the [docs][]
* [ ] Appropriate Helm chart updates in the [helm-charts][]

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/
[helm-charts]: https://github.com/open-telemetry/opentelemetry-helm-charts

### CODE REVIEW

**Review by julianocosta89 — APPROVED** [review]
Awesome!

**Review by rogercoll — COMMENTED** [review]
LGTM, just a small question

File: .gitignore [hunk 1]
```diff
@@ -54,4 +54,3 @@ test/tracetesting/tracetesting-vars.yaml
 /src/paymentservice/demo.proto
 /src/shippingservice/proto/
 /src/currencyservice/proto
-/src/accountingservice/genproto
```
[the remark this answers is not in the document]
Comment (julianocosta89) [reply .gitignore:57]: ah yes, we have discussed that in the SIG meeting yesterday. We do still need it!

[the remark this answers is not in the document]
Comment (julianocosta89) [reply .gitignore:57]: Added it back.
The folder is not called `genproto` though 🤔

**Comment by puckpuck** [comment]
Ahh... technically we should of left the genproto stuff off. It was left over from when the accountingservice originally started as a go based service before being moved to .NET.

I'll write another PR to remove it.

**Pull request merged**